Print Kani version

adpaco-aws opened the pull request:
### Description of changes: 

Dusting this off since it'd be great to have for next release.

This PR changes Kani so it prints its version and the invocation type/mode as
`Kani Rust Verifier <version> (<invocation>)` where `<invocation>` is either `standalone` or `cargo plugin`.

Please note that I'm leaving development version information for later in #2617. We can continue the discussion we started in #2571 here.

### Resolved issues:

Resolves #2570 

### Testing:

* How is this change tested? Two new tests.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

celinval commented on kani-driver/build.rs:
```diff
@@ -3,11 +3,28 @@
+}
+
+/// Retrieve the `kani-verifier` version using `cargo metadata`
+fn kani_verifier_version() -> String {
```
Shouldn't the `kani-driver` version always match `kani-verifier`? I don't understand why we are doing this.

adpaco-aws replied:
I've opened #2626 to ensure these are the same versions. We just use the CARGO_PKG_VERSION now.

celinval commented on kani-driver/src/main.rs:
```diff
@@ -77,6 +82,11 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
     }
 
+    if !session.args.common_args.quiet {
+        let kani_version = print_kani_version();
```
Shouldn't this method print the version? It has `print` in the name. :)

adpaco-aws replied:
True, it prints now.

celinval commented on kani-driver/src/version.rs:
```diff
@@ -0,0 +1,30 @@
+const KANI_RUST_VERIFIER: &str = "Kani Rust Verifier";
+const KANI_VERIFIER_VERSION: &str = env!("KANI_VERIFIER_VERSION");
+
+/// Print Kani version. At present, this is only release version information.
```
This doesn't print though

celinval commented on kani-driver/src/version.rs:
```diff
@@ -0,0 +1,30 @@
+    let mut version_str = "Kani Rust Verifier ".to_string();
+    version_str.push_str(KANI_VERIFIER_VERSION);
+
+    let invocation_str = match determine_invocation_type(Vec::from_iter(std::env::args_os())) {
```
nit: you could get this as an argument.

adpaco-aws replied:
Done.

celinval reviewed (commented):
Shouldn't we print the same blob for all sub-commands as well?

adpaco-aws commented:
Addressed all comments.

> Shouldn't we print the same blob for all sub-commands as well?

To be honest, this wasn't clear to me. Guessing it's because I'd really like `assess` to be a tool rather than a sub-command...

celinval reviewed (approved):
Thanks

celinval commented on kani-driver/src/version.rs:
```diff
@@ -0,0 +1,34 @@
+///  - `<invocation>` is `cargo plugin` if Kani was invoked with `cargo kani` or
+///    `standalone` if it was invoked with `kani`.
+fn kani_version_release(invocation_type: InvocationType) -> String {
+    let mut version_str = "Kani Rust Verifier ".to_string();
```
do you ever use this?

adpaco-aws replied:
Good catch. I'm surprised the compiler/`clippy` didn't complain about this.
